### PR TITLE
Add CI runners matrix (ubuntu 20.04, 22.04)

### PR DESCRIPTION
**Pull request by yankay**
Upgrade the CI environment by adding the ubuntu 22.04 to the CI.

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
/lgtm

**Comment by stmcginnis** [comment]
/lgtm
/retest

Failing tests are for the `cgroupv2` workflow in setting up the Fedora Vagrant box. So shouldn't have anything to do with this change.

File: .github/workflows/docker.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ permissions:
 jobs:
   docker:
     name: Docker
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
```
[the remark this answers is not in the document]
Comment (aojea) [reply .github/workflows/docker.yaml:17]: that is a good suggestion, is prow using cgroupsv2 @BenTheElder ?

[the remark this answers is not in the document]
Comment (BenTheElder) [reply .github/workflows/docker.yaml:17]: > that is a good suggestion, is prow using cgroupsv2 @BenTheElder ?

AFAIK the GKE clusters are not yet, though it's possible to do that. https://github.com/kubernetes/k8s.io/issues/5276
I'm not sure for the EKS cluster.

[the remark this answers is not in the document]
Comment (BenTheElder) [reply .github/workflows/docker.yaml:17]: I don't think we should depend on which cgroups the prowjobs are on.

[the remark this answers is not in the document]
Comment (AkihiroSuda) [reply .github/workflows/docker.yaml:17]: ping @yankay

[the remark this answers is not in the document]
Comment (yankay) [reply .github/workflows/docker.yaml:17]: Thanks @AkihiroSuda 

The PR is rebased :-)

[the remark this answers is not in the document]
Comment (AkihiroSuda) [reply .github/workflows/docker.yaml:17]: Too early to drop cgroup v1

[the remark this answers is not in the document]
Comment (yankay) [reply .github/workflows/docker.yaml]: Thanks @AkihiroSuda 

The code has been changed to keep Ubuntu 20.04 and add Ubuntu 22.04 to "docker CI."

**Review by AkihiroSuda — APPROVED** [review]
Thanks

File: .github/workflows/docker.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (aojea) [reply .github/workflows/docker.yaml]: What's the final state? Ubuntu 20.04 equal to.cgroupsv1 testing and 22.04 to cgroupsv2 ?

If that's the case can we make it more evident?

[the remark this answers is not in the document]
Comment (aojea) [reply .github/workflows/docker.yaml]: Ok, nevermind, I'll figure it out later

[the remark this answers is not in the document]
Comment (aojea) [reply .github/workflows/docker.yaml]: But this makes.the ci to.run 14 jobs on presubmit, seems excessive

**Comment by aojea** [comment]
I´m going to close these as this will make the presubmits too expensive to maintain (in terms of support and time to run) for any PR

**Pull request closed without merging**